---
title: "FS9 report"
author: "Created by: K. T. Mou"
date: "Updated on: 6Apr2021"
output:
  html_document:
    toc: true
    toc_float: true
    number_sections: true
---
***
# mothur
**Goal**: Process 16S rRNA sequence data and generate output for R scripts.
1. Used mothur version 1.40.1, silva release version 132 taxonomy, and fasta files.
  - silva.nr_v132.tax
  - silva.nr_v132.pcr.align (renamed as silva.v4.fasta)
  - *.fastq
2. Run the following commands in same directory where fastq files are. 
3. Load mothur program.

<details><summary>Mothur commands</summary>
```{r echo=TRUE, eval=FALSE}
make.file(inputdir=/project/fastq/, type=fastq, prefix=stability)

make.contigs(file=stability.files, processors=40)

summary.seqs(fasta=stability.trim.contigs.fasta)

screen.seqs(fasta=stability.trim.contigs.fasta, maxambig=0, maxlength=275, maxhomop=6, group=stability.contigs.groups)

summary.seqs(fasta=current)

unique.seqs(fasta=stability.trim.contigs.good.fasta)

count.seqs(name=current, group=stability.contigs.good.groups)

summary.seqs(count=current, fasta=current, processors=39)

pcr.seqs(fasta=silva.nr_v132.align, start=11894, end=25319, keepdots=F)

rename.file(input=silva.nr_v132.pcr.align, new=silva.v4.fasta)

summary.seqs(fasta=silva.v4.fasta)

align.seqs(fasta=stability.trim.contigs.good.unique.fasta, reference=silva.v4.fasta, flip=T)

summary.seqs(fasta=current, count=stability.trim.contigs.good.count_table)

screen.seqs(fasta=stability.trim.contigs.good.unique.align, count=stability.trim.contigs.good.count_table, start=1968, end=11550, maxhomop=6)

summary.seqs(fasta=current, count=current)

filter.seqs(fasta=stability.trim.contigs.good.unique.good.align, vertical=T, trump=.)

unique.seqs(fasta=stability.trim.contigs.good.unique.good.filter.fasta, count=stability.trim.contigs.good.good.count_table)

pre.cluster(fasta=current, count=current, diffs=2)

chimera.vsearch(fasta=current, count=current, dereplicate=t)

remove.seqs(fasta=current, accnos=current)

summary.seqs(fasta=current, count=current)

classify.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table, reference=silva.v4.fasta, taxonomy=silva.nr_v132.tax, cutoff=80)

remove.lineage(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table, taxonomy=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.taxonomy, taxon=Chloroplast-Mitochondria-unknown-Archaea-Eukaryota)

summary.tax(taxonomy=current, count=current)

get.groups(count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, groups=Mock5)

seq.error(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, reference=mock_community_16S3.fna, aligned=F)

#Rename these files to use for split.abund command (remove sequences with 2 or fewer occurrences)
#rename.file(input=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, new=stability.outdoubletons.fasta)
#rename.file(input=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, new=stability.outdoubletons.count_table)

#Remove Mock group
remove.groups(count=stability.outdoubletons.count_table, fasta=stability.outdoubletons.fasta, taxonomy=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.pick.taxonomy, groups=Mock5)

#Split out doubletons from fasta and count table data (they'll be put in the "rare" files)
split.abund(fasta=stability.outdoubletons.fasta, count=stability.outdoubletons.count_table, cutoff=2, accnos=true)

dist.seqs(fasta=stability.outdoubletons.abund.fasta, cutoff=0.03)

cluster(column=current, count=stability.outdoubletons.abund.count_table)

make.shared(list=current, count=stability.outdoubletons.abund.count_table, label=0.03)

classify.otu(list=stability.outdoubletons.abund.opti_mcc.list, count=stability.outdoubletons.abund.count_table, taxonomy=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.pick.taxonomy, label=0.03)

count.groups(shared=stability.outdoubletons.abund.opti_mcc.shared)

sub.sample(shared=stability.outdoubletons.abund.opti_mcc.shared, size=1300)
```
</details>

## Prepare OTU table
1. Create R Project.
2. Files needed to make OTU table:
- stability.outdoubletons.abund.opti_mcc.0.03.cons.taxonomy
  - Save "stability.outdoubletons.abund.opti_mcc.0.03.cons.taxonomy" as a csv file in a spreadsheet editor
- stability.outdoubletons.abund.opti_mcc.0.03.subsample.shared
  - Save "stability.outdoubletons.abund.opti_mcc.0.03.subsample.shared" as a csv file in a spreadsheet editor; remove "label", and "numOtus" rows
3. Run the files through the following R script.

<details><summary>1_OTUtable.R script</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages
library(tidyverse)

#########################################

#Edit taxonomy file
taxonomy <- read.csv("~/FS9/data/stability.outdoubletons.abund.opti_mcc.0.03.cons.taxonomy.csv")
taxonomy
taxonomy$Taxonomy <- gsub('*\\(.*?\\) *', '', taxonomy$Taxonomy) #Remove (100) from "Taxonomy" column
taxonomy
taxonomy[1:6,3] #Show Taxonomy column rows 1 through 6
write.csv(taxonomy, file = "FS9.taxonomy.doubleton.csv") #In excel, remove the "Size" and numbered rownames

#Edit subsample.shared file
shared <- read.csv("~/FS9/data/stability.outdoubletons.abund.opti_mcc.0.03.subsample.shared.csv")
head(shared)
shared[1:6,1]
shared <- t(shared) #Transpose
head(shared)
shared[1:6,1]
write.csv(shared, file = 'FS9.subsample.shared.doubleton.csv') #Open in a spreadsheet editor and remove the "V*" row

#Read new subsample.shared file and edit column name
shared <- read.csv("~/FS9/data/FS9.subsample.shared.doubleton.csv")
head(shared)
colnames(shared) [1] <- "OTU" #Rename first column to "OTU"

#Read new taxonomy file and merge with 'shared' dataframe
taxonomy <- read.csv("~/FS9/data/FS9.taxonomy.doubleton.csv")
OTUtable <- merge(shared, taxonomy, by.x ="OTU", by.y = "OTU") #Merge 'shared' and 'taxonomy' via "OTU"
head(OTUtable)
nrow(OTUtable) #1583 (singletons removed), 1405 (doubletons removed)
ncol(OTUtable) #170 (singletons removed), 169 (doubletons removed)
write.csv(OTUtable, file= "FS9.OTUtable.doubleton.csv") #In a spreadsheet editor, open "FS9.OTUtable.doubleton.csv" and remove first column (unnecessary)

#check OTU table
OTUtable <-read.csv("~/FS9/data/FS9.OTUtable.doubleton.csv")
head(OTUtable)
```
</details>

***
# phyloseq
**Goal**: Generate phyloseq object to use for 3a_alpha_beta_diversity_NONINFnm_INFnm.R and 3b_alpha_beta_diversity_INFnmfeedinject.R. Run adonis function with distance matrices to assess how variation is attributed to different experimental treatments or uncontrolled covariates.

## NONINFnm vs INFnm
<details><summary>2a_phyloseq_NONINFnm_INFnm.R</summary>
```{r echo=TRUE, eval=FALSE}
meta <- read.csv("~/FS9/data/FS9_metadata_NONINFnm_vs_INFnm.csv", row.names = 1)
otu <- read.csv("~/FS9/data/FS9.OTUtable.doubleton.csv", row.names=1)
dim(otu) #1405 168
head(otu[,165:168])

#Remove taxonomy from 'otu'
tax <- otu[,(167:168)] #Removed column 168 "Taxonomy" and 167 to include the row names
head(tax)
colnames(tax)[1] <- "delete" #Renamed column 1 (formerly 167) as "delete" which will later be deleted
head(tax)

#Modify 'otu' with only OTU count data
otu <- otu[,-168] #Remove column 168 "Taxonomy" to have only OTU data
head(otu[,165:167]) 
dim(otu) #1405 rows 167 columns

#Transpose 'otu' to match format of 'meta'
otu.trans <- t(otu) #Now rownames are sample names, columns are OTUs
head(otu.trans[,1:5])
class(meta) #dataframe
class(otu) #dataframe

#Merge 'otu' and 'meta' data frames
otu.meta <- merge(meta, otu.trans, by.x=0, by.y=0) 
#x=0 means match via rownames from 'meta'; y=0 means match via rownames from 'otu.trans'
head(otu.meta[,1:10])
dim(otu.meta) #45 1410 (doubletons removed)
rownames(otu.meta) <- otu.meta[,1] #Set column 1 as rownames for 'otu.meta'
otu.meta <- otu.meta[,-1]
class(otu.meta) #dataframe
rownames(otu.meta)

#Added an "All" column (combines 'Treatment' and 'Day' values) in new 'otu.meta2'
otu.meta2<- cbind(otu.meta) #Make second copy of otu.meta to use to include "All" column
colnames(otu.meta2)
otu.meta2$All <- with(otu.meta2, paste0(Day, sep="_", Treatment)) #Create "All" column with Day and Treatment combined
head(otu.meta2)
dim(otu.meta2) #45 1410 (doubletons removed)
head(otu.meta2[,1405:1410])
head(otu.meta2[,1:10])
otu.meta2<- otu.meta2[,c(1:4,1410,5:1409)] #Reorder columns to have "All" column after "Treatment" column
#write.csv(otu.meta2, file="FS9.otu.meta_all.doubleton.csv")

#Pull out metadata from 'otu.meta2' dataframe
head(otu.meta2[,1:10])
meta2 <- otu.meta2[,c(1:5)] #Take columns 1-5 ("Day" to "All") from 'otu.meta2' to make 'meta2'
head(meta2)
dim(meta2) #45 5 (doubletons removed)

#Create SAM metadata table phyloseq object
SAM = sample_data(meta2, errorIfNULL = TRUE)
head(SAM)
dim(SAM) #45 5

#Pull out OTU data from 'otu.meta2' dataframe
head(otu.meta2[,1:10])
tail(otu.meta2[,1405:1410])
otu2 <- otu.meta2[,c(6:1410)] #Select OTU columns to create 'otu2' dataframe
head(otu2[,1:10])
dim(otu2) #45 1405 (doubletons removed)
otu2.trans <- t(otu2) #Transpose otu.all to have OTUs as rownames, sample names as column names
head(otu2.trans[,1:10])
dim(otu2.trans) #1405 45 (doubletons removed)

#Merge 'tax' back into 'otu2.trans' for correct format and taxons
head(tax)
otu2.tax <- merge(otu2.trans, tax, by=0) #Merge by rownames aka OTU rownames
dim(otu2.tax) #1405 48 (doubletons removed)
head(otu2.tax[,1:10])
head(otu2.tax[,40:48])
row.names(otu2.tax) <- otu2.tax[,1] #Set first row of 'otu2.tax' as rownames
head(otu2.tax[,1:5])
otu2.tax <- otu2.tax[,-1] #Remove first row aka extraneous "Row.names" column from 'otu2.tax'
head(otu2.tax[,1:5])

#Split again
dim(otu2.tax) #1405 47 (doubletons removed)
head(otu2.tax[,40:47])
otu2.notax <- otu2.tax[,1:45] #take rows 1-29 to make new dataframe 'otu2.notax' (46 is "delete" column, 47 is "Taxonomy" column)
dim(otu2.notax) #1405 45 (doubletons removed)
head(otu2.notax[,1:5])
head(otu2.notax[,40:45])
class(otu2.notax) #dataframe
otu2.notax <- as.matrix(otu2.notax) #turn 'otu2.notax' into a matrix class
class(otu2.notax) #matrix
otu2.notax
otu2.notax.trans <- t(otu2.notax)
head(otu2.notax.trans[,1:10])

#Create OTU table phyloseq object
OTU = otu_table(otu2.notax.trans, taxa_are_rows = FALSE)
head(OTU[,1:10])
dim(OTU) #45 1405 (doubletons removed)
class(OTU)
OTU2 <- prune_taxa(taxa_sums(OTU) > 0, OTU)
class(OTU2)
head(OTU2[,1:10])
taxa_sums(OTU2)
dim(OTU2) #45 901 (doubletons removed)

#Edit taxonomy
dim(otu2.tax) #1405 47 (doubletons removed)
head(otu2.tax[,40:47])
tax2 <- separate(data = otu2.tax, 
                 col = Taxonomy, 
                 into=c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";")
#"separate" function separates "Taxonomy" column into 7 separate columns labeled "Kingdom", "Phylum", "Class", etc.
head(tax2) #notice that Species column is blank
dim(tax2) #1405 53 (doubletons removed)
head(tax2[,45:53])
tax2.kg <- tax2[,47:52] #Keep only taxonomy columns "Kingdom" to "Genus"
head(tax2.kg)
dim(tax2.kg) #1405 6 (doubletons removed)
class(tax2.kg) #dataframe
tax2.kg <- as.matrix(tax2.kg)
class(tax2.kg) #matrix
tax2.kg

#Create TAX taxonomy table phyloseq object
TAX = tax_table(tax2.kg)
head(TAX)
dim(TAX) #1405 6 (doubletons removed)

#Create phyloseq object containing taxonomy, metadata, and otu table
phyloseq.FS9 <- phyloseq(OTU2, SAM, TAX) #prune out any OTUs that have total to 0 in all samples
phyloseq.FS9
#otu_table()   OTU Table:         [ 901 taxa and 45 samples ]
#sample_data() Sample Data:       [ 45 samples by 5 sample variables ]
#tax_table()   Taxonomy Table:    [ 901 taxa by 6 taxonomic ranks ]

#save(phyloseq.FS9, file="phyloseq.FS9.doubleton.RData") #Use for FS9_alpha_beta_diversity.R when loading phyloseq.FS9 object

#Distance calculation
dist.FS9 <- distance(phyloseq.FS9, method="bray")
phyloseq.adonis <- as(sample_data(phyloseq.FS9), "data.frame")
set.seed(1)
adonis.FS9 <- adonis(dist.FS9~Day*Treatment, data=phyloseq.adonis, permutations=9999)
adonis.FS9
```

```{r echo=FALSE, eval=TRUE, warning=FALSE, message=FALSE}
#Load library packages
library(vegan)
library(tidyverse)
library(phyloseq)

#########################################

meta <- read.csv("~/FS9/data/FS9_metadata_NONINFnm_vs_INFnm.csv", row.names = 1)
otu <- read.csv("~/FS9/data/FS9.OTUtable.doubleton.csv", row.names=1)

#Remove taxonomy from 'otu'
tax <- otu[,(167:168)] #Removed column 168 "Taxonomy" and 167 to include the row names
colnames(tax)[1] <- "delete" #Renamed column 1 (formerly 167) as "delete" which will later be deleted

#Modify 'otu' with only OTU count data
otu <- otu[,-168] #Remove column 168 "Taxonomy" to have only OTU data

#Transpose 'otu' to match format of 'meta'
otu.trans <- t(otu) #Now rownames are sample names, columns are OTUs

#Merge 'otu' and 'meta' data frames
otu.meta <- merge(meta, otu.trans, by.x=0, by.y=0) 
#x=0 means match via rownames from 'meta'; y=0 means match via rownames from 'otu.trans'
rownames(otu.meta) <- otu.meta[,1] #Set column 1 as rownames for 'otu.meta'
otu.meta <- otu.meta[,-1]

#Added an "All" column (combines 'Treatment' and 'Day' values) in new 'otu.meta2'
otu.meta2<- cbind(otu.meta) #Make second copy of otu.meta to use to include "All" column
otu.meta2$All <- with(otu.meta2, paste0(Day, sep="_", Treatment)) #Create "All" column with Day and Treatment combined
otu.meta2<- otu.meta2[,c(1:4,1410,5:1409)] #Reorder columns to have "All" column after "Treatment" column
#write.csv(otu.meta2, file="FS9.otu.meta_all.doubleton.csv")

#Pull out metadata from 'otu.meta2' dataframe
meta2 <- otu.meta2[,c(1:5)] #Take columns 1-5 ("Day" to "All") from 'otu.meta2' to make 'meta2'

#Create SAM metadata table phyloseq object
SAM = sample_data(meta2, errorIfNULL = TRUE)

#Pull out OTU data from 'otu.meta2' dataframe
otu2 <- otu.meta2[,c(6:1410)] #Select OTU columns to create 'otu2' dataframe
otu2.trans <- t(otu2) #Transpose otu.all to have OTUs as rownames, sample names as column names

#Merge 'tax' back into 'otu2.trans' for correct format and taxons
otu2.tax <- merge(otu2.trans, tax, by=0) #Merge by rownames aka OTU rownames
row.names(otu2.tax) <- otu2.tax[,1] #Set first row of 'otu2.tax' as rownames
otu2.tax <- otu2.tax[,-1] #Remove first row aka extraneous "Row.names" column from 'otu2.tax'

#Split again
otu2.notax <- otu2.tax[,1:45] #take rows 1-29 to make new dataframe 'otu2.notax' (46 is "delete" column, 47 is "Taxonomy" column)
otu2.notax <- as.matrix(otu2.notax) #turn 'otu2.notax' into a matrix class
otu2.notax.trans <- t(otu2.notax)

#Create OTU table phyloseq object
OTU = otu_table(otu2.notax.trans, taxa_are_rows = FALSE)
OTU2 <- prune_taxa(taxa_sums(OTU) > 0, OTU)

#Edit taxonomy
tax2 <- separate(data = otu2.tax, 
                 col = Taxonomy, 
                 into=c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";")
#"separate" function separates "Taxonomy" column into 7 separate columns labeled "Kingdom", "Phylum", "Class", etc.
tax2.kg <- tax2[,47:52] #Keep only taxonomy columns "Kingdom" to "Genus"
tax2.kg <- as.matrix(tax2.kg)

#Create TAX taxonomy table phyloseq object
TAX = tax_table(tax2.kg)

#Create phyloseq object containing taxonomy, metadata, and otu table
phyloseq.FS9 <- phyloseq(OTU2, SAM, TAX) #prune out any OTUs that have total to 0 in all samples
#otu_table()   OTU Table:         [ 901 taxa and 45 samples ]
#sample_data() Sample Data:       [ 45 samples by 5 sample variables ]
#tax_table()   Taxonomy Table:    [ 901 taxa by 6 taxonomic ranks ]

#save(phyloseq.FS9, file="phyloseq.FS9.doubleton.RData") #Use for FS9_alpha_beta_diversity.R when loading phyloseq.FS9 object

#Distance calculation
dist.FS9 <- distance(phyloseq.FS9, method="bray")
phyloseq.adonis <- as(sample_data(phyloseq.FS9), "data.frame")
set.seed(1)
adonis.FS9 <- adonis(dist.FS9~Day*Treatment, data=phyloseq.adonis, permutations=9999)
adonis.FS9
```
</details>
  
## INFnm vs INFfeed vs INFinject
<details><summary>2b_phyloseq_INFnmfeedinject.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```

```{r echo=TRUE, eval=TRUE, warning=FALSE, message=FALSE}
#Load library packages

#########################################

```
</details>

***
# alpha and beta diversity
**Goal**:

## NONINFnm vs INFnm
<details><summary>3a_alpha_beta_diversity_NONINFnm_INFnm.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```
</details>

## INFnm vs INFfeed vs INFinject
<details><summary>3b_alpha_beta_diversity_INFnmfeedinject.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```
</details>

***
# DESeq2 
**Goal**:

## NONINFnm vs INFnm
<details><summary>4a_DeSeq2_NONINFnm_INFnm.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```
</details>

## INFnm vs INFfeed vs INFinject
<details><summary>4b_DeSeq2_INFnmfeedinject.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```
</details>

***
## Figure 4 for INFnm vs INFfeed vs INFinject
**Goal**:

<details><summary>5b_NMDS_DeSeq2_INFnmfeedinject.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```
</details>

***
# Abx
**Goal**:
<details><summary>AbxConcWeightADGLungLesion.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```
</details>

***
# qPCR
**Goal**:
<details><summary>AMR_qPCR.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages

#########################################

```
</details>